### PR TITLE
Yatiyaya BEGONE check_surrounding_area port

### PR DESCRIPTION
**Pull request by cdb-is-not-good**
Wholecloth ports some improvements from Liberty station courtesy of yati

https://github.com/Liberty-Landing/Liberty-Station-13/pull/446
https://github.com/Liberty-Landing/Liberty-Station-13/pull/455

Compiles, seems to work, no runtimes AFAIK. Review ASAP, pls.

## About The Pull Request
<details>
<summary>
	About The Pull Request
</summary>
<hr>

	
<hr>
</details>

## Changelog
:cl:

/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/superior_animal/superior_animal.dm [hunk 1]
```diff
@@ -591,22 +589,17 @@
 		handle_regular_status_updates() // we should probably still do this even if we're dead or something
 		ticks_processed = 0
 
-	if (!weakened)
-
-		if(!AI_inactive) //we dont need to handle ai if we're disabled
-			handle_ai()
-			//Speaking
+	if (!weakened || !AI_inactive) // OR...
```
Comment (king5327) [inline code/modules/mob/living/carbon/superior_animal/superior_animal.dm:592]: This shouldn't be an actual optimization unless the compiler be shit.

File: code/modules/sanity/sanity_mob.dm [hunk 2]
```diff
@@ -153,7 +153,7 @@ GLOBAL_VAR_INIT(GLOBAL_INSIGHT_MOD, 1)
 
 /datum/sanity/proc/handle_view()
 	. = 0
-	activate_mobs_in_range(owner, SANITY_MOB_DISTANCE_ACTIVATION)
+	activate_mobs_in_range(owner, world.view)
```
Comment (king5327) [inline code/modules/sanity/sanity_mob.dm:156]: This isn't an optimization, it just reduces the search area, which might be undesirable.

Comment (king5327) [reply code/modules/sanity/sanity_mob.dm:156]: Can we redefine SANITY_MOB_DISTANCE_ACTIVATION to world.view instead of its current 12 instead?

**Comment by cdb-is-not-good** [comment]
After some testing this didn't seem to offer serious benefits over the wonkiness it introduced in passive 'farm' mobs like chickens, etc.

**Pull request closed without merging**